ui/app/selectors/custom-gas.js

```diff
@@ -206,30 +206,17 @@ export function getGasPriceInHexWei (price) {
   return addHexPrefix(priceEstimateToWei(value))
 }
 
-export function getRenderableBasicEstimateData (state, gasLimit, useFastestButtons) {
-  if (getBasicGasEstimateLoadingStatus(state)) {
-    return []
-  }
-
-  const { showFiatInTestnets } = getPreferences(state)
-  const isMainnet = getIsMainnet(state)
-  const showFiat = (isMainnet || Boolean(showFiatInTestnets))
-  const { conversionRate } = state.metamask
-  const currentCurrency = getCurrentCurrency(state)
+export function getRenderableGasButtonData (estimates, gasLimit, showFiat, conversionRate, currentCurrency) {
```

Comment:
So is this function now used to transform the gas basic estimate state, which is shared between the `swaps` slice gas estimates and the `gas` slice?

Assuming my understanding here is correct, I don't think this belongs in this module anymore. This module is for selectors for the `gas` slice, and this is neither a selector nor is it used solely for the `gas` slice.

Reply:
Your understanding is correct. How would you feel about fixing this in another PR? I raise this because moving this function brings forth a host of other issues that should be resolved at the same time, but doing so would expand the scope of this PR.

`getRenderableGasButtonData` depends on four other functions that are defined in and exported from the same file: `getRenderableEthFee`, `getRenderableConvertedCurrencyFee`, `getRenderableTimeEstimate`, `getGasPriceInHexWei`. Each of these is used by at least one other method in the same file.

If I move `getRenderableGasButtonData` out of this module, we then really should move each of these out of this module, for reasons similar to those given in your original comment. Moreover, each of these can be simplified and perhaps entirely replaced by one or two methods imported from our conversion utilities. This would be a nice solution because we can just delete these dependencies of `getRenderableGasButtonData`

But I'd prefer not to do all that in this PR. And moving `getRenderableGasButtonData` will feel incomplete.

Reply:
Sure, dealing with this later seems OK

Reply:
I will create a github issue for this

- [ ] Create github issue for refactoring location of getRenderableGasButtonData and use of dependencies

Reply:
Created a ticket for this: https://github.com/MetaMask/metamask-extension/issues/9713